### ISSUE TITLE
Message.fromId is null问题依旧存在 pat消息似乎没有起到作用

### ISSUE THREAD

**Issue by fangmuke** [wechaty/puppet-padlocal#64]
- wechaty: `0.56.6`
- wechaty-puppet: `0.34.3`
- wechaty-puppet-padlocal: `0.2.42`

拍一拍消息
```1|bot  | WechatifiedMessage {
1|bot  |   _events: [Object: null prototype] {},
1|bot  |   _eventsCount: 0,
1|bot  |   _maxListeners: undefined,
1|bot  |   id: '5587541694366639755',
1|bot  |   payload: {
1|bot  |     id: '5587541694366639755',
1|bot  |     timestamp: 1618143402,
1|bot  |     type: 0,
1|bot  |     fromId: undefined,
1|bot  |     mentionIdList: [],
1|bot  |     roomId: '18839872723@chatroom',
1|bot  |     text: '"PHP2Java(PSL" 拍了拍 "泡泡茶壶的迷弟(java"',
1|bot  |     toId: undefined
1|bot  |   },
1|bot  |   [Symbol(kCapture)]: false
1|bot  | }
1|bot  | Error: payload.fromId is null?
1|bot  |     at WechatifiedMessage.talker (/root/bot/node_modules/wechaty/src/user/message.ts:268:13)
1|bot  |     at WechatifiedMessage.toString (/root/bot/node_modules/wechaty/src/user/message.ts:206:12)
1|bot  |     at Array.join (<anonymous>)
1|bot  |     at Array.toString (<anonymous>)
1|bot  |     at String (<anonymous>)
1|bot  |     at Wechaty.<anonymous> (/root/bot/node_modules/wechaty-plugin-contrib/src/contrib/event-logger.ts:34:75)
1|bot  |     at Wechaty.emit (events.js:327:22)
1|bot  |     at Wechaty.EventEmitter.emit (domain.js:483:12)
1|bot  |     at PuppetPadlocal.<anonymous> (/root/bot/node_modules/wechaty/src/wechaty.ts:483:18)
1|bot  |     at runMicrotasks (<anonymous>)
1|bot  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

添加管理员消息
```
1|bot  | WechatifiedMessage {
1|bot  |   _events: [Object: null prototype] {},
1|bot  |   _eventsCount: 0,
1|bot  |   _maxListeners: undefined,
1|bot  |   id: '1589113500620971495',
1|bot  |   payload: {
1|bot  |     id: '1589113500620971495',
1|bot  |     timestamp: 1618143514,
1|bot  |     type: 13,
1|bot  |     fromId: undefined,
1|bot  |     mentionIdList: [],
1|bot  |     roomId: '19481319952@chatroom',
1|bot  |     text: '19481319952@chatroom:\n' +
1|bot  |       '<sysmsg type="sysmsgtemplate">\n' +
1|bot  |       '\t<sysmsgtemplate>\n' +
1|bot  |       '\t\t<content_template type="tmpl_type_admin_explain">\n' +
1|bot  |       '\t\t\t<plain><![CDATA[]]></plain>\n' +
1|bot  |       '\t\t\t<template><![CDATA[你已经被群主"$ownername$"设置为群管理员 $adminexplain$]]></template>\n' +
1|bot  |       '\t\t\t<link_list>\n' +
1|bot  |       '\t\t\t\t<link name="ownername" type="link_profile">\n' +
1|bot  |       '\t\t\t\t\t<memberlist>\n' +
1|bot  |       '\t\t\t\t\t\t<member>\n' +
1|bot  |       '\t\t\t\t\t\t\t<username><![CDATA[wxid_vdha0qorca4e22]]></username>\n' +
1|bot  |       '\t\t\t\t\t\t\t<nickname><![CDATA[test]]></nickname>\n' +
1|bot  |       '\t\t\t\t\t\t</member>\n' +
1|bot  |       '\t\t\t\t\t</memberlist>\n' +
1|bot  |       '\t\t\t\t</link>\n' +
1|bot  |       '\t\t\t\t<link name="adminexplain" type="link_admin_explain">\n' +
1|bot  |       '\t\t\t\t\t<title><![CDATA[查看详情]]></title>\n' +
1|bot  |       '\t\t\t\t</link>\n' +
1|bot  |       '\t\t\t</link_list>\n' +
1|bot  |       '\t\t</content_template>\n' +
1|bot  |       '\t</sysmsgtemplate>\n' +
1|bot  |       '</sysmsg>\n',
1|bot  |     toId: undefined
1|bot  |   },
1|bot  |   [Symbol(kCapture)]: false
1|bot  | }
1|bot  | Error: payload.fromId is null?
1|bot  |     at WechatifiedMessage.talker (/root/bot/node_modules/wechaty/src/user/message.ts:268:13)
1|bot  |     at WechatifiedMessage.toString (/root/bot/node_modules/wechaty/src/user/message.ts:206:12)
1|bot  |     at Array.join (<anonymous>)
1|bot  |     at Array.toString (<anonymous>)
1|bot  |     at String (<anonymous>)
1|bot  |     at Wechaty.<anonymous> (/root/bot/node_modules/wechaty-plugin-contrib/src/contrib/event-logger.ts:34:75)
1|bot  |     at Wechaty.emit (events.js:327:22)
1|bot  |     at Wechaty.EventEmitter.emit (domain.js:483:12)
1|bot  |     at PuppetPadlocal.<anonymous> (/root/bot/node_modules/wechaty/src/wechaty.ts:483:18)
1|bot  |     at runMicrotasks (<anonymous>)
1|bot  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

完成群待办消息
```
1|bot  | TypeError: Cannot read property 'sysmsg' of undefined
1|bot  |     at Object.isPatMessage (/root/bot/node_modules/wechaty-puppet-padlocal/src/padlocal/message-parser/helpers/message-pat.ts:39:17)
1|bot  |     at runMicrotasks (<anonymous>)
1|bot  |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
1|bot  |     at Object.padLocalMessageToWechaty (/root/bot/node_modules/wechaty-puppet-padlocal/src/padlocal/schema-mapper/index.ts:58:28)
```

`拍一拍消息`、`添加管理员消息` 是`wechaty-plugin-contrib`的`event-logger`插件中调用了`talker()`造成的异常。
`完成群待办消息`是pat类型消息造成的。

**Comment by fangmuke** [comment]
我们是不是应该增加一些消息类型？ @huan

**Comment by huan** [comment]
A proposal for adding new message types will be welcome for discussion!

**Comment by padlocal** [comment]
I suggest to add  "Pat" as the new type in [MessageType]( https://github.com/wechaty/wechaty-puppet/blob/2132e7514cef8e38564e97b808dee8748f116e19/src/schemas/message.ts#L1)

```ts
export enum MessageType {
  ...
  Pat = 16,
}
```

Is it okay?

**Comment by huan** [comment]
> I suggest adding "Pat" as the new type in MessageType

Could you please file an issue for it so that we can continue discussing this creating new MessageType conversation based on it?

**Comment by paulhybryant** [comment]
My bot is crashing from time to time due to such errors. 
Any plan to fix it or at least surface the error so that it can be try-catched?

**Comment by padlocal** [comment]
Fixed in` wechaty-puppet-padlocal@1.11.18`
